resources/frontend.rb

```diff
@@ -19,8 +19,19 @@
       variables['frontend'][new_resource.name] ||= {}
       variables['frontend'][new_resource.name]['default_backend'] ||= ''
       variables['frontend'][new_resource.name]['default_backend'] << new_resource.default_backend
-      variables['frontend'][new_resource.name]['bind'] ||= ''
-      variables['frontend'][new_resource.name]['bind'] << new_resource.bind
+      if new_resource.bind.is_a? Hash
+        variables['frontend'][new_resource.name]['bind'] = []
+        new_resource.bind.map do |addresses, ports|
+          addresses = Array(addresses) unless addresses.is_a? Array
+          ports = Array(ports) unless ports.is_a? Array
+          (Array(addresses).product Array(ports)).each do |combo|
+            variables['frontend'][new_resource.name]['bind'] << "#{combo.join(':')}"
```

Comment:
You should have this be `combo.join(':')` no need to wrap it in a string w/ interpolation.  `join` already returns a string.